### ISSUE TITLE
Distributing commercially vs MIT license

### ISSUE THREAD

**Issue by isaacs** [DotMail/DotMail#25]
Hi.  First of all, I have no intent to distribute DotMail, commercially or otherwise.  Also, IANAL.  Just a friendly neighborhood open source citizen trying to do the right thing.

In your readme, it states: 

```
DotMail is available under the MIT open source license. See LICENSE.md for more details.

Please don’t distribute commercially without prior consent!
```

But in the LICENSE file (a standard MIT license), it states:

> **Permission is hereby granted**, free of charge, to any person obtaining a copy of
> this software and associated documentation files (the "Software"), to deal in
> the Software without restriction, including without limitation the rights to
> use, copy, modify, merge, **publish, distribute, sublicense, and/or sell** copies of
> the Software, and to permit persons to whom the Software is furnished to do so,
> subject to the following conditions:

(Emphasis mine)

So, it seems to me that "prior consent" has already been granted to commercially distribute the software.  This specific consent was explicitly granted, free of charge, in writing, to any person obtaining a copy of the software.

It seems to me that either you're using the wrong license (maybe a flavor of GPL would be more appropriate, with a note that you're willing to discuss other licenses for commercial entities?) or the request might be oddly phrased (maybe you are ok with commercial distribution of the code and docs, as long as copyright attribution is clear and the "DotMail" name is not infringed?)

As it stands, though, it's kind of confusing.

**Comment by isaacs** [comment]
Also, you may be interested in the Artistic 2.0 license that npm uses.  It allows almost all uses, including commercial uses, but does not allow users to charge a license fee for the software itself.  https://github.com/npm/npm/blob/master/LICENSE

**Comment by CodaFi** [comment]
Honestly, I wrote that because I don't want people uploading a million unfinished copies of this to the App Store.  I really don't care what happens with the code beyond that (it's certainly got nothing to do with money).

**Comment by isaacs** [comment]
Well... your license currently allows this, though you have a claim to a trademark on the name "DotMail" (which you should register with the USPTO if you're serious about it.  Again, IANAL, but probably you should talk to one that does IP stuff.)

Maybe it would be worthwhile to just call out that it's unfinished, and to please not publish to the App Store?  You could also take a look at other licenses, or maybe say that the code is MIT, but some other parts (build scripts, assets, etc) are "all rights reserved" or something?

**Comment by CodaFi** [comment]
Yeah, throwing the assets under NPM sounds like a good plan.  If you have a wording in mind for the README and license file, go ahead and submit a pull request.
